### PR TITLE
Integrate fetch with frontend catalogue

### PR DESCRIPTION
**Pull request by binkymilk**
Tasks:
- [x] Load catalog items using the API
- [x] Load the catalog item page using the API
- [x] Fix loading and error screens
- [x] Fix `evaluationMetrics` type mismatch

### CODE REVIEW

File: src/api/index.ts [hunk 1]
```diff
@@ -2,8 +2,15 @@ import type { CatalogueItemType } from 'types/CatalogueItem.type'
 
 // eslint-disable-next-line import/prefer-default-export
 export const fetchCatalogItems = async (): Promise<CatalogueItemType[]> => {
-	const response = await fetch('api/data/catalog.json') // Replace with your own JSON file path
+	const response = await fetch('/api/data/catalog.json') // Replace with your own JSON file path
```
Comment (binkymilk) [inline src/api/index.ts:5]: Added a / to fix API path when calling in the catalogue item page

File: src/api/index.ts [hunk 2]
```diff
@@ -2,8 +2,15 @@ import type { CatalogueItemType } from 'types/CatalogueItem.type'
...
 	const catalog: CatalogueItemType[] = await response.json()
 	return catalog
 }
+
+export const fetchCatalogItem = async (
+	catalogItemId?: string
+): Promise<CatalogueItemType | undefined> => {
+	const catalogItems: CatalogueItemType[] = await fetchCatalogItems()
+	return catalogItems.find(item => item.id === catalogItemId)
+}
```
Comment (binkymilk) [inline src/api/index.ts:16]: @butchtm I added a faux-API endpoint to get a specific catalog item given the id. Let me know what's your recommendation to handle this and filtering!

**Comment by binkymilk** [comment]
@butchtm I also noticed a change in the `catalog.json` schema, particularly the `evaluation-metrics` part. Are we expecting an array of objects that could be a `{"metric": <obj>}` or a `{"link": <obj>}`? 

I was thinking it could be better if we standardize the types in the array if we're expecting more than one entry for this.

Option 1:
```
evaluationMetric: {
  metric: { metric-type: '', value: 0 },
  link: { url: '', description: '' }
}
```

Option 2:
```
evaluationMetric: [
  { metric: { metric-type: '', value: 0 }, link: { url: '', description: '' } },
  { metric: { metric-type: '', value: 0 }, link: { url: '', description: '' } },
  { metric: { metric-type: '', value: 0 }, link: { url: '', description: '' } }
]
```

Lmk what you think!

**Comment by binkymilk** [comment]
> @butchtm I also noticed a change in the `catalog.json` schema, particularly the `evaluation-metrics` part. Are we expecting an array of objects that could be a `{"metric": <obj>}` or a `{"link": <obj>}`?
> 
> I was thinking it could be better if we standardize the types in the array if we're expecting more than one entry for this.
> 
> Option 1 (if we're expecting one metric only):
> 
> ```
> evaluationMetric: {
>   metric: { metric-type: '', value: 0 },
>   link: { url: '', description: '' }
> }
> ```
> 
> Option 2 (if we're expecting multiple metrics):
> 
> ```
> evaluationMetric: [
>   { metric: { metric-type: '', value: 0 }, link: { url: '', description: '' } },
>   { metric: { metric-type: '', value: 0 }, link: { url: '', description: '' } },
>   { metric: { metric-type: '', value: 0 }, link: { url: '', description: '' } }
> ]
> ```
> 
> Lmk what you think!

Never mind this! I found a simple workaround in the front-end. I also just realized that there won't be any issues in the data entry side because it's in yaml rather than json.

File: src/api/index.ts (hunk 2 above)
Comment (butchtm) [reply src/api/index.ts:16]: Yes! this is exactly the same implem I'm going for.

**Review by butchtm — APPROVED** [review]
Will rebase my pending changes based on this PR instead

**Pull request merged**